### DISCUSSION TITLE
Kubeapps UI sometimes fails due to the kubeapps-postgresql Liveness and Readiness probes

### DISCUSSION THREAD

**Discussion by JoeKuan** [kubeapps/kubeapps#4588] [Q&A] [answered]
I am experimenting minikube + kubapps on my 2015 MBP and I notice the Kubeapps UI sometimes fails to respond.

After spending a bit of time debugging, I found out it's the kubeapps-postgresql causing the error with Liveness and Readiness probles. Here are the events log for the kubeapps-postgresql-primary-0:

Events:
  Type     Reason     Age                 From     Message
  ----     ------     ----                ----     -------
  Warning  Unhealthy  41m (x25 over 62m)  kubelet  Readiness probe failed: 127.0.0.1:5432 - no response
  Warning  BackOff    36m (x13 over 54m)  kubelet  Back-off restarting failed container
  Warning  Unhealthy  27m (x63 over 65m)  kubelet  Readiness probe failed: command "/bin/sh -c -e exec pg_isready -U \"postgres\" -d \"dbname=assets\" -h 127.0.0.1 -p 5432\n[ -f /opt/bitnami/postgresql/tmp/.initialized ] || [ -f /bitnami/postgresql/.initialized ]\n" timed out
  Warning  Unhealthy  12m (x52 over 65m)  kubelet  Liveness probe failed: command "/bin/sh -c exec pg_isready -U \"postgres\" -d \"dbname=assets\" -h 127.0.0.1 -p 5432" timed out

My guess is the laptop not beefy enough to pass the probe tests.

**Comment by absoludity** [comment, marked as answer]
> My guess is the laptop not beefy enough to pass the probe tests.

Could be - though I'd expect the actual logs of the pod to show you that too. Note, if you're running on your local machine, I usually use something like this manifest to reduce the resource usage:

https://github.com/vmware-tanzu/kubeapps/blob/main/docs/howto/manifests/kubeapps-local-dev-values.yaml

Just take the relevant bits from there to ensure postgres runs in standalone mode, and you only get a single pod for each deployment etc.
